mypy/modulefinder.py

```diff
@@ -387,13 +387,13 @@ def find_modules_recursive(self, module: str) -> List[BuildSource]:
                     if mod not in hits:
                         hits.add(mod)
                         result += self.find_modules_recursive(module + '.' + mod)
-        elif os.path.isdir(module_path) and module in self.ns_packages:
-            # Even more subtler: handle recursive decent into PEP 420
+        elif os.path.isdir(module_path) and any(module.startswith(p) for p in self.ns_packages):
```

Comment:
I wonder if startswith is correct here. For example, consider `foobar` vs `foo`?

Reply:
Good point, but now that I'm looking at the check, I think it's unnecessary (always true, and running tests with an assert seems to confirm that).